src/geometry.rs

```diff
@@ -244,6 +250,138 @@ impl Geometry {
         }
         used_normals.len() == self.normals().len()
     }
+
+    /// Calculates topological relations (neighborhood) of mesh face -> faces.
+    /// Returns a Map (key: face index, value: list of its neighboring faces indices)
+    pub fn face_to_face_topology(&self) -> HashMap<usize, SmallVec<[usize; 16]>> {
+        let mut f2f: HashMap<usize, SmallVec<[usize; 16]>> = HashMap::new();
+        for (from_counter, f) in self.triangle_faces_iter().enumerate() {
```

Comment:
I know we only have triangle faces so far, but once we add e.g. quads, this will continue to compile but will no longer be correct. Shouldn't we iterate `self.faces` instead so we are forced to handle all kinds of faces?

Also applies everywhere else this is used in this PR.

Reply:
True that but i also expect you'll want to put all the topology-computing functions outside the `Geometry` and this is the only face getter right now. Shall we add a general face getter, jut like the one for vertices?